### PR TITLE
Bootstrap test-proxy service

### PR DESCRIPTION
**Pull request by heaths**
Relates to #1874 but does not complete it.

### CODE REVIEW

**Comment by heaths** [comment]
@RickWinter @analogrelay I'm still working on the transport but nearly done. I figured enough was here already I should push the bootstrapping portion (minus automatic acquisition, tracked in #1955 so I can get an MVP out by EOW) to alleviate some of the review burden. I'll `rebase --onto` whatever gets merged here.

File: sdk/core/azure_core_test/src/proxy.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+pub async fn start(
+    test_data_dir: impl AsRef<Path>,
+    options: Option<ProxyOptions>,
+) -> Result<Proxy> {
+    if env::var(PROXY_MANUAL_START).is_ok_and(|v| v.to_ascii_lowercase() == "true") {
+        return Err(azure_core::Error::message(
+            ErrorKind::Other,
+            format!("environment variable {PROXY_MANUAL_START} is 'true'; not starting test proxy"),
+        ));
```
Comment (analogrelay) [inline sdk/core/azure_core_test/src/proxy.rs:33]: Should this be an error? If a user has manually started the proxy and then wants to run a test against the manually-started proxy, wouldn't this cause their test to fail, assuming there is code calling `proxy.start()?`?

File: sdk/core/azure_core_test/src/sanitizers.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        json_path: String::from("$.values"),
+        value: None,
+        regex: None,
+        group_for_replace: None,
+    };
+    let headers = sut.as_headers().expect("expect headers");
+    headers.for_each(|(h, v)| {
+        assert_eq!(h.as_str(), "x-abstraction-identifier");
+        assert_eq!(v.as_str(), "BodyKeySanitizer");
+    });
```
Comment (analogrelay) [inline sdk/core/azure_core_test/src/sanitizers.rs:81]: nit: This won't fail if `as_headers` returns empty or an iterator with 15 `x-abstraction-identifier: BodyKeySanitizer` headers, if I'm reading right. I'd prefer a `.collect::<Vec<_>>()` and then assert the collected vector to ensure we're asserting the exact content of the iterator.

File: sdk/core/azure_core_test/src/sanitizers.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+impl_sanitizer!(BodyRegexSanitizer);
+
+#[test]
+fn test_body_regex_sanitizer_as_headers() {
+    let sut = BodyRegexSanitizer::default();
+    let headers = sut.as_headers().expect("expect headers");
+    headers.for_each(|(h, v)| {
+        assert_eq!(h.as_str(), "x-abstraction-identifier");
+        assert_eq!(v.as_str(), "BodyRegexSanitizer");
+    });
```
Comment (analogrelay) [inline sdk/core/azure_core_test/src/sanitizers.rs:117]: Same here.

**Review by analogrelay — APPROVED** [review]
One small issue with tests on the sanitizers, but I trust you to take care of it, so nothing blocking on my end.

File: sdk/core/azure_core_test/src/sanitizers.rs (hunk 2 above)
Comment (heaths) [reply sdk/core/azure_core_test/src/sanitizers.rs:81]: Good catch.

File: sdk/core/azure_core_test/src/proxy.rs (hunk 1 above)
Comment (heaths) [reply sdk/core/azure_core_test/src/proxy.rs:33]: You're right. I incorrect transcribed the Go source (the intent, rather; Go isn't the problem at all) that I pulled this from.

**Review by analogrelay — APPROVED** [review]

**Pull request merged**